core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -7148,6 +7377,36 @@ private static class NonLiteralException extends RuntimeException {
         private static final NonLiteralException INSTANCE = new NonLiteralException();
     }
 
+    private class ChildTableColumnFinder implements PostOrderTreeTraversalAlgo.Visitor {
+        private boolean found;
+        private QueryModel targetModel;
+
+        public ChildTableColumnFinder(QueryModel targetModel, boolean found) {
+            this.targetModel = targetModel;
+            this.found = found;
+        }
+
+        @Override
+        public void visit(ExpressionNode node) throws SqlException {
+            if (node.type == LITERAL) {
+                CharSequence clauseColumn = node.token;
+                int dot = Chars.indexOfLastUnquoted(clauseColumn, '.');
+                CharSequence clauseColumnName = clauseColumn.subSequence(dot + 1, clauseColumn.length());
+                //check with column in between clause should be of master table
+                if (dot != -1) {
+                    CharSequence whereClauseAlias = clauseColumn.subSequence(0, dot);
+                    CharSequence masterTableAlias = targetModel.getAlias() != null ? targetModel.getAlias().token :
+                            targetModel.getTableNameExpr().token;
+                    if (!Chars.equals(whereClauseAlias, masterTableAlias))
```

Comment:
Names in SQL are case-insensitive, so `Chars#equalsIgnoreCase()` should be used for such comparisons.